Modules/overlapped.c

```diff
@@ -708,8 +773,41 @@ Overlapped_getresult(OverlappedObject *self, PyObject *args)
             if (transferred != PyBytes_GET_SIZE(self->allocated_buffer) &&
                 _PyBytes_Resize(&self->allocated_buffer, transferred))
                 return NULL;
+
             Py_INCREF(self->allocated_buffer);
             return self->allocated_buffer;
+        case TYPE_READ_FROM:
+            assert(PyBytes_CheckExact(self->read_from.buffer));
+
+            if (transferred != PyBytes_GET_SIZE(self->read_from.buffer) &&
+                _PyBytes_Resize(&self->read_from.buffer, transferred))
+            {
+                return NULL;
+            }
+
+            // unparse the address
+            addr = unparse_address((SOCKADDR*)&self->read_from.address,
+                                   self->read_from.address_length);
+
+            if (addr == NULL) {
+                return NULL;
+            }
+
+            // The result is a two item tuple: (message, (address, port))
+            self->read_from.result = PyTuple_New(2);
+            if (self->read_from.result == NULL) {
+                return NULL;
```

Comment:
Doesn't this leak a reference to `addr`?

Reply:
Yup, good catch.